src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -153,6 +144,7 @@
         this.callbackRunner = Executors.newSingleThreadExecutor();
 
         this.executor = options.getExecutor();
+        this.connectExecutor = Executors.newSingleThreadExecutor();

```

Comment:
Do we want to shutdown the `connectExecutor` during `close` similar to the `callbackRunner`?